app/controllers/admin/problems_controller.rb

```diff
@@ -13,17 +13,17 @@ def index
   def create
     authorize contest, :edit?   
     params[:problem][:category_ids] ||= []
-    @problem = contest.problems.build(params.require(:problem).permit!.except(:archive, :solution))
+    problem = contest.problems.build(params.require(:problem).permit!.except(:archive, :solution))
```

Comment:
You can't avoid using `@problem` here. This action is doing something different thing, so it should continue using `@problem`